From: Dan Williams <dan.j.williams@intel.com>
Date: Wed, 6 Jan 2016 12:03:41 -0800
Subject: libnvdimm, pmem: move definition of nvdimm_namespace_add_poison to nd.h
Git-commit: ad9a8bde2cb19f6876f964fc48acc8b6a2f325ff
Patch-mainline: v4.5-rc1
References: FATE#319858, FATE#320457

nd-core.h is private to the libnvdimm core internals and should not be
used by drivers.

Signed-off-by: Dan Williams <dan.j.williams@intel.com>
Acked-by: Johannes Thumshirn <jthumshirn@suse.com>
---
 drivers/nvdimm/nd-core.h |    2 --
 drivers/nvdimm/nd.h      |    2 ++
 drivers/nvdimm/pmem.c    |    1 -
 3 files changed, 2 insertions(+), 3 deletions(-)

--- a/drivers/nvdimm/nd-core.h
+++ b/drivers/nvdimm/nd-core.h
@@ -91,6 +91,4 @@ bool __nd_attach_ndns(struct device *dev
 ssize_t nd_namespace_store(struct device *dev,
 		struct nd_namespace_common **_ndns, const char *buf,
 		size_t len);
-int nvdimm_namespace_add_poison(struct gendisk *disk, resource_size_t offset,
-		struct nd_namespace_common *ndns);
 #endif /* __ND_CORE_H__ */
--- a/drivers/nvdimm/nd.h
+++ b/drivers/nvdimm/nd.h
@@ -262,6 +262,8 @@ int nvdimm_namespace_attach_btt(struct n
 int nvdimm_namespace_detach_btt(struct nd_namespace_common *ndns);
 const char *nvdimm_namespace_disk_name(struct nd_namespace_common *ndns,
 		char *name);
+int nvdimm_namespace_add_poison(struct gendisk *disk, resource_size_t offset,
+		struct nd_namespace_common *ndns);
 int nd_blk_region_init(struct nd_region *nd_region);
 void __nd_iostat_start(struct bio *bio, unsigned long *start);
 static inline bool nd_iostat_start(struct bio *bio, unsigned long *start)
--- a/drivers/nvdimm/pmem.c
+++ b/drivers/nvdimm/pmem.c
@@ -27,7 +27,6 @@
 #include <linux/slab.h>
 #include <linux/pmem.h>
 #include <linux/nd.h>
-#include "nd-core.h"
 #include "pfn.h"
 #include "nd.h"
 
